### DISCUSSION TITLE
Automatically load all necessary namespaces when switching the locale in typesafe-i18n/vue

### DISCUSSION THREAD

**Discussion by nkoehring** [ivanhofer/typesafe-i18n#535] [Show and tell]
The following GIST contains a composable for Vue3, that triggers loading of all namespaces given in the route meta information. This allows abstracting the locale namespaces out of the component into the page definition. It also triggers reload on locale change, which otherwise would have to be reimplemented inside each component that uses namespaces.

https://gist.github.com/nkoehring/08531494ef97d0d36fb82630c2481e5e

Please let me know in case there is already a better way to do this.

**Comment by ivanhofer** [comment]
Looks good! I did something similar in a recent `SvelteKit` project of mine.
Maybe we can add some kind of "recipes" section to the bottom of the [`adapter-vue` docs](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/adapter-vue)?

**Comment by nkoehring** [comment]
Good idea! Feel free to use the code any way you like.

**Reply by ivanhofer** [reply]
Would you like to create a PR for that?
It just needs a small description of what problem it solves and maybe some comments inside the code to better understand which parts do what.
I recently added a ["recipes" section](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/adapter-svelte#recipes) to the `Svelte` docs.
